atom/browser/osr/osr_render_widget_host_view.cc

```diff
@@ -909,128 +859,70 @@ void OffScreenRenderWidgetHostView::SetNeedsBeginFrames(
   begin_frame_timer_->SetActive(needs_begin_frames);
 
   if (software_output_device_) {
-    software_output_device_->SetActive(needs_begin_frames && painting_, false);
-  }
-}
-
-void CopyBitmapTo(const SkBitmap& destination,
-                  const SkBitmap& source,
-                  const gfx::Rect& pos) {
-  char* src = static_cast<char*>(source.getPixels());
-  char* dest = static_cast<char*>(destination.getPixels());
-  int pixelsize = source.bytesPerPixel();
-
-  int width =
-      pos.x() + pos.width() <= destination.width()
-          ? pos.width()
-          : pos.width() - ((pos.x() + pos.width()) - destination.width());
-  int height =
-      pos.y() + pos.height() <= destination.height()
-          ? pos.height()
-          : pos.height() - ((pos.y() + pos.height()) - destination.height());
-
-  if (width > 0 && height > 0) {
-    for (int i = 0; i < height; i++) {
-      memcpy(dest + ((pos.y() + i) * destination.width() + pos.x()) * pixelsize,
-             src + (i * source.width()) * pixelsize, width * pixelsize);
-    }
+    software_output_device_->SetActive(painting_, false);
   }
-
-  destination.notifyPixelsChanged();
 }
 
 void OffScreenRenderWidgetHostView::OnPaint(const gfx::Rect& damage_rect,
                                             const SkBitmap& bitmap) {
   TRACE_EVENT0("electron", "OffScreenRenderWidgetHostView::OnPaint");
 
-  HoldResize();
-
   if (parent_callback_) {
     parent_callback_.Run(damage_rect, bitmap);
   } else {
     gfx::Rect damage(damage_rect);
+    gfx::Canvas canvas(size(), 1.0f, false);
 
-    std::vector<gfx::Rect> damages;
-    std::vector<const SkBitmap*> bitmaps;
-    std::vector<SkBitmap> originals;
+    canvas.DrawImageInt(gfx::ImageSkia::CreateFrom1xBitmap(bitmap), 0, 0);
 
     if (popup_host_view_ && popup_bitmap_.get()) {
       gfx::Rect pos = popup_host_view_->popup_position_;
       damage.Union(pos);
-      damages.push_back(pos);
-      bitmaps.push_back(popup_bitmap_.get());
-      originals.push_back(SkBitmapOperations::CreateTiledBitmap(
-          bitmap, pos.x(), pos.y(), pos.width(), pos.height()));
+      canvas.DrawImageInt(
+          gfx::ImageSkia::CreateFrom1xBitmap(*popup_bitmap_.get()), 0, 0,
+          pos.width(), pos.height(), pos.x(), pos.y(), pos.width(),
+          pos.height(), false);
     }
 
-    for (auto* proxy_view : proxy_views_) {
-      gfx::Rect pos = proxy_view->GetBounds();
+    for (auto proxy_view : proxy_views_) {
```

Comment:
i think dropping the `*` from `auto*` will trip up the chromium-style checker (which currently isn't actually running in the gyp build...)